### PR TITLE
Implement List and Apply for rds configurations

### PR DESCRIPTION
**Pull request by lego963**
Add possibility to List parameter config

Add possibility to Apply some config to instances

```
=== RUN   TestConfigurationsList
--- PASS: TestConfigurationsList (2.80s)
=== RUN   TestConfigurationsLifecycle
--- PASS: TestConfigurationsLifecycle (3.46s)
=== RUN   TestConfigurationsApply
--- PASS: TestConfigurationsApply (643.02s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: openstack/rds/v3/configurations/results.go [hunk 1]
```diff
@@ -49,6 +57,28 @@ type Parameter struct {
...
+	ConfigurationID string `json:"configuration_id"`
+	// Specifies the parameter template name.
+	ConfigurationName string `json:"configuration_name"`
+	// Specifies the result of applying the parameter template.
+	ApplyResults []ApplyConfigurationResults `json:"apply_results"`
+	// Specifies whether each parameter template is applied to DB instances successfully.
+	Success bool `json:"success"`
+}
+
+type ApplyConfigurationResults struct {
```
Comment (outcatcher) [inline openstack/rds/v3/configurations/results.go:71]: Should be `ApplyConfigurationResult`, not `...Results`

File: openstack/rds/v3/configurations/results.go [hunk 2]
```diff
@@ -78,14 +112,42 @@ func (r GetResult) Extract() (*Configuration, error) {
 	return &response, err
 }
 
+// Extract is a function that accepts a result and extracts a list of configurations.
+func (lr ListResult) Extract() ([]Configuration, error) {
```
Comment (outcatcher) [inline openstack/rds/v3/configurations/results.go:116]: The reciever is named `r` normally. See [naming](https://github.com/opentelekomcloud/gophertelekomcloud/blob/master/STYLEGUIDE.md#naming) guide

File: openstack/rds/v3/configurations/results.go [hunk 3]
```diff
@@ -60,10 +90,14 @@ func (r CreateResult) ExtractInto(v interface{}) error {
 	return r.Result.ExtractIntoStructPtr(v, "configuration")
 }
 
+type commonResult struct {
```
Comment (outcatcher) [inline openstack/rds/v3/configurations/results.go:93]: There is no real sense to use `commonResult` ― there are no methods bind to the type.

File: openstack/rds/v3/configurations/results.go [hunk 4]
```diff
@@ -78,14 +112,42 @@ func (r GetResult) Extract() (*Configuration, error) {
...
+func (lr ListResult) Extract() ([]Configuration, error) {
+	var a struct {
+		Configurations []Configuration `json:"configurations"`
+	}
+	err := lr.Result.ExtractInto(&a)
+	return a.Configurations, err
+}
+
+// Extract is a function that accepts a result and extracts an apply configuration result.
+func (ar ApplyResult) Extract() (*ApplyConfiguration, error) {
```
Comment (outcatcher) [inline openstack/rds/v3/configurations/results.go:125]: The method should preferably be after the type definition

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by lego963** [comment]
regate

**Comment by lego963** [comment]
recheck

**Comment by lego963** [comment]
recheck

**Pull request merged**